pkg/sdk/external_volumes_def.go

```diff
@@ -0,0 +1,161 @@
+	QueryStructField(
+		"Encryption",
+		g.NewQueryStruct("ExternalVolumeS3Encryption").
+			Assignment("TYPE", g.KindOfT[S3EncryptionType](), g.ParameterOptions().SingleQuotes().Required()).
+			OptionalTextAssignment("KMS_KEY_ID", g.ParameterOptions().SingleQuotes()),
+		g.ListOptions().Parentheses().NoComma().SQL("ENCRYPTION =").Required(),
+	)
+
+var externalGCSStorageLocationDef = g.NewQueryStruct("GCSStorageLocationParams").
+	TextAssignment("NAME", g.ParameterOptions().SingleQuotes().Required()).
+	PredefinedQueryStructField("StorageProviderGcs", "string", g.StaticOptions().SQL("STORAGE_PROVIDER = 'GCS'")).
+	TextAssignment("STORAGE_BASE_URL", g.ParameterOptions().SingleQuotes().Required()).
+	QueryStructField(
+		"Encryption",
+		g.NewQueryStruct("ExternalVolumeGCSEncryption").
+			Assignment("TYPE", g.KindOfT[GCSEncryptionType](), g.ParameterOptions().SingleQuotes().Required()).
+			OptionalTextAssignment("KMS_KEY_ID", g.ParameterOptions().SingleQuotes()),
+		g.ListOptions().Parentheses().NoComma().SQL("ENCRYPTION =").Required(),
+	)
+
+var externalAzureStorageLocationDef = g.NewQueryStruct("AzureStorageLocationParams").
+	TextAssignment("NAME", g.ParameterOptions().SingleQuotes().Required()).
+	PredefinedQueryStructField("StorageProviderAzure", "string", g.StaticOptions().SQL("STORAGE_PROVIDER = 'AZURE'")).
+	TextAssignment("AZURE_TENANT_ID", g.ParameterOptions().SingleQuotes().Required()).
+	TextAssignment("STORAGE_BASE_URL", g.ParameterOptions().SingleQuotes().Required())
+
+var storageLocationDef = g.NewQueryStruct("ExternalVolumeStorageLocation"). // Can't name StorageLocation due to naming clash with type in storage integration
+										OptionalQueryStructField(
+		"S3StorageLocationParams",
+		externalS3StorageLocationDef,
+		g.ListOptions().Parentheses().NoComma(),
+	).
+	OptionalQueryStructField(
+		"GCSStorageLocationParams",
+		externalGCSStorageLocationDef,
+		g.ListOptions().Parentheses().NoComma(),
+	).
+	OptionalQueryStructField(
+		"AzureStorageLocationParams",
+		externalAzureStorageLocationDef,
+		g.ListOptions().Parentheses().NoComma(),
+	).
+	WithValidation(g.ExactlyOneValueSet, "S3StorageLocationParams", "GCSStorageLocationParams", "AzureStorageLocationParams")
+
+var ExternalVolumesDef = g.NewInterface(
+	"ExternalVolumes",
+	"ExternalVolume",
+	g.KindOfT[AccountObjectIdentifier](),
+).
+	CreateOperation(
+		"https://docs.snowflake.com/en/sql-reference/sql/create-external-volume",
+		g.NewQueryStruct("CreateExternalVolume").
+			Create().
+			OrReplace().
+			SQL("EXTERNAL VOLUME").
+			IfNotExists().
+			Name().
+			ListAssignment("STORAGE_LOCATIONS", "ExternalVolumeStorageLocation", g.ParameterOptions().Parentheses().Required()).
+			OptionalBooleanAssignment("ALLOW_WRITES", nil).
+			OptionalComment().
+			WithValidation(g.ConflictingFields, "OrReplace", "IfNotExists").
+			WithValidation(g.ValidIdentifier, "name"),
+		storageLocationDef,
+	).
+	AlterOperation(
```

Comment:
General question: I see that there are no ALTER UNSET statements inside the docs. Did you try running them manually? I am asking because they are essential for the proper terraform resource design:
- If they are missing, we know we have to have workarounds inside the resource, and we will approach the team to ask for the addition
- If they are just not documented, we can reach out to the doc team and ask for a change.

Reply:
Just tried running `alter external volume test_external_volume_new unset comment;` and got error `Unsupported feature 'UNSET for external volume parameters'.`.

Reply:
ah, that's what I was afraid of. Thanks for checking, we will reach out to the proper team internally and ask for the addition.